src/runtime/server/utils/rate-limit.ts

```diff
@@ -1,76 +1,65 @@
-import { H3Event, getRequestHeader } from 'h3'
-import { useRuntimeConfig } from '#imports'
-import type { RateLimit, RateLimitRoutes } from '../../../types'
+import {H3Event, getRequestHeader, getRequestURL} from 'h3'
+import { getRouteRules } from '#imports'
+import type { RateLimit } from '../../../types'
+import {RouteRateLimitOptions} from "../../../types";
 
 // store rate limits for each IP address and URL
 const rateLimit: RateLimit = {}
```

Comment:
One thing not included in this PR but that you should consider, is supporting useStorage with configurable destination.

This would allow people to sync their route rules across distributed applications.